File: python-nodes/linear/06_prepare_next_loop.py
# Node 6: Prepare Next Loop
# Input:  order + receipt from Node 5
# Output: summary + loop_payload sent back to Node 1 via webhook

results = []

for item in _items:
    data = item["json"]
    cycle = int(data.get("cycle", 0))
    next_cycle = cycle + 1

    results.append(
        {
            "json": {
                "cycle": cycle,
                "next_cycle": next_cycle,
                "last_order_id": data.get("order_id"),
                "last_final_total": data.get("final_total"),
                "last_receipt": data.get("receipt"),
                "stage": "loop_ready",
                # This payload becomes the input for Node 1 on the next run.
                "loop_payload": {
                    "cycle": next_cycle,
                    "running": True,
                },
            }
        }
    )

return results

File: javascript-nodes/linear/06_prepare_next_loop.js
// Node 6: Prepare Next Loop
// Input: order + receipt from Node 5  →  Output: loop_payload for Node 1

const results = [];

for (const item of $input.all()) {
  const data = item.json;
  const cycle = Number(data.cycle ?? 0);
  const next_cycle = cycle + 1;

  results.push({
    json: {
      cycle,
      next_cycle,
      last_order_id: data.order_id,
      last_final_total: data.final_total,
      last_receipt: data.receipt,
      stage: 'loop_ready',
      loop_payload: {
        cycle: next_cycle,
        running: true,
      },
    },
  });
}

return results;
